src/renderer/components/layout/sidebar.tsx

```diff
@@ -191,17 +212,25 @@ export class Sidebar extends React.Component<Props> {
             >
               {this.renderCustomResources()}
             </SidebarNavItem>
-            {clusterPageMenuRegistry.getItems().map(({ title, target, components: { Icon } }) => {
-              const registeredPage = clusterPageRegistry.getByPageMenuTarget(target);
-              if (!registeredPage) return;
-              const { extensionId, id: pageId } = registeredPage;
-              const pageUrl = getExtensionPageUrl({ extensionId, pageId, params: target.params });
-              const isActive = pageUrl === navigation.location.pathname;
+            {clusterPageMenuRegistry.getRootItems().map((menuItem) => {
+              const registeredPage = clusterPageRegistry.getByPageMenuTarget(menuItem.target);
+              let pageUrl: string
+              let isActive = false
+              if (registeredPage) {
+                const { extensionId, id: pageId } = registeredPage;
+                pageUrl = getExtensionPageUrl({ extensionId, pageId, params: menuItem.target.params });
+                isActive = pageUrl === navigation.location.pathname;
+              }
+              const tabRoutes = this.getTabLayoutRoutes(menuItem)
+              if (!registeredPage && tabRoutes.length == 0) {
+                return
+              }
               return (
                 <SidebarNavItem
                   key={pageUrl} url={pageUrl}
-                  text={title} icon={<Icon/>}
+                  text={menuItem.title} icon={<menuItem.components.Icon/>}
                   isActive={isActive}
+                  subMenus={tabRoutes}
                 />
               );
             })}
```

Comment:
Last thing, probably minor. But this is getting rather long so (imo) it should be in its own render function. Maybe `renderRegisteredMenus`?

Reply:
If not, I will approve

Reply:
Fixed.